ui/src/pages/project.tsx

```diff
@@ -69,8 +84,23 @@ export const Project = ({
 
       <Pipelines
         project={data?.result?.value as _Project}
+        stageName={stageName}
+        promotionId={promotionId}
+        promote={
+          freight && stage
+            ? {
+                freight,
+                stage
+              }
+            : undefined
+        }
         creatingStage={creatingStage}
         creatingWarehouse={creatingWarehouse}
+        warehouseName={warehouseName}
+        freightName={freightName}
+        warehouses={listWarehousesQuery.data?.warehouses || []}
+        freights={getFreightQuery.data?.groups || {}}
+        refetchFreights={getFreightQuery.refetch}
```

Comment:
What's the point of passing all those params? Let's get url params directly in the component, and fetch data there too. We don't need a loading state here.

Reply:
Its either pollute pipelines or project so I chose project due to the fact that it would be easy to split from here for pages in future if we ever want. I kept loading state here because I want data ready before it renders for details

Reply:
So now we have two loaders, one here and the second one in the pipeline. It doesn't look good.
If there is no need to have that data now in the project component, please move everything to the pipeline component.

Reply:
I can move stage loader up to project but the sole reason of having those in project component is to have data ready pipeline component on first attempt

Reply:
I checked this file one more time, and I don't see any reason to pass 11 params if we can have everything inside the pipeline component.